erizo/src/erizo/WebRtcConnection.cpp

```diff
@@ -741,4 +741,22 @@ void WebRtcConnection::setTransport(std::shared_ptr<Transport> transport) {  //
   bundle_ = true;
 }
 
+std::pair<RTPExtensionsMap, RTPExtensionsMap> WebRtcConnection::getSourceExtensionMap() {
+  std::lock_guard<std::mutex> lk(extension_map_mutex);
+  auto result = std::make_pair(extension_processor_.getVideoExtensionMap(),
+                               extension_processor_.getAudioExtensionMap());
+  return result;
+}
+
+void WebRtcConnection::setSourceExtensionMap(std::shared_ptr<WebRtcConnection> source_wrtc) {
+  asyncTask([source_wrtc](std::shared_ptr<WebRtcConnection> this_connection){
+    auto source_map_future = source_wrtc->worker_->post_async_task([source_wrtc]{
+      return source_wrtc->getSourceExtensionMap();
+    });
+    auto map = source_map_future.get();
```

Comment:
I think we wouldn't need  the usage of the mutex or at least I can't think of a case where we access that info from different threads.
Anyway, you're blocking thread execution here. Another option would be:
```js
std::weak_ptr<WebRtcConnection> weak_this = shared_from_this();
source_wrtc->asyncTask([weak_this](std::shared_ptr<WebRtcConnection> source){
    map = source->getSourceExtensionMap();
    if (auto this_ptr = weak_this.lock()) {
        this_ptr->asyncTask([map](std::shared_ptr<WebRtcConnection> this_connection) {
            this_connection->setSourceExtensionMap(map);
        });
    }
});
```
It's just pseudo code so it will have errors, but it's just to illustrate the idea. By doing this we would get rid of the new post_async_task and the `mutex` (I always really try to remove them from this context).